internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -152,16 +161,29 @@ func controlPlaneUpdateStatusWithAvailableVersions(currentClusterVersion *versio
 	}, nil
 }
 
-func workerUpdateStatus(currentClusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node) (NodeVersionInfoUpdate, error) {
-	return workerUpdateStatusWithAvailableVersions(currentClusterVersion, allNodesVersioningInfo, node, kubernetes.StaticVersionInquirer{})
+func workerUpdateStatus(clusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node) (NodeVersionInfoUpdate, error) {
+	return workerUpdateStatusWithAvailableVersions(clusterVersion, allNodesVersioningInfo, node, kubernetes.StaticVersionInquirer{})
 }
 
-func workerUpdateStatusWithAvailableVersions(currentClusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node, versionInquirer kubernetes.VersionInquirer) (NodeVersionInfoUpdate, error) {
+func workerUpdateStatusWithAvailableVersions(clusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node, versionInquirer kubernetes.VersionInquirer) (NodeVersionInfoUpdate, error) {
+	// Checking worker nodes for updates is a bit different than checking a control plane node.
+	// It can be that an upgrade has already been started on the control plane
+	// or that all nodes are still on the same version (no upgrade started yet).
+	// First we need to check if an upgrade has aready been started
+	// This can be determined by kubernetes.AllNodesMatchClusterVersion(allNodesVersioningInfo, clusterVersion)
+	allNodesMatchCurrentClusterVersion := kubernetes.AllNodesMatchClusterVersionWithVersioningInfo(allNodesVersioningInfo, clusterVersion)
 	// Check that all control plane nodes have at least the current cluster version we plan to
 	// upgrade this worker node to. If not, they need to be fully upgraded first
-	controlPlanesMatchVersion := kubernetes.AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo, currentClusterVersion)
-	if !controlPlanesMatchVersion {
-		return NodeVersionInfoUpdate{}, errors.New("at least one control plane does not tolerate the current cluster version")
+	controlPlanesMatchVersion := kubernetes.AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo, clusterVersion)
+	// Check if there is a newer version
+	versionCompare, err := clusterVersion.Compare(kubernetes.LatestVersion().String())
```

Comment:
Can this have side effects in a case like:

* Current cluster version: `1.13.x`
* Latest version: `1.15.2`
* Versions to go through: `1.13.x -> 1.14.y -> 1.15.2`

I think with this check we'll never be able to upgrade worker nodes from `1.13.x -> 1.14.y` because `LatestVersion()` reports `1.15.2`.

Reply:
Nevermind, it's fine, `versionCompare < 0 && (allNodesMatchCurrentClusterVersion || !controlPlanesMatchVersion)` should do the right thing.